buildscripts/Set-VersionInfo.ps1

```diff
@@ -0,0 +1,58 @@
+<#
+
+.SYNOPSIS
+
+Extracts build version information from Settings.proj and the
+environment. Then constructs several flavor version strings and
+uses them, along with the current year to
+
+  - update buildscripts/CommonAssemblyInfo.cs,
+  - update all project.json files, and
+  - emit TeamCity directives so it can know about the values
+
+#>
+
+[xml] $settings = Get-Content 'Settings.proj'
+$version_group = $settings.Project.PropertyGroup[1]
+
+$major = $version_group.Project_Major
+$minor = $version_group.Project_Minor
+$build = $version_group.Project_Build
+
+if ( $env:release_build -eq "true" )
+{
+    $version_suffix = ""
+}
+else
+{
+    $version_suffix = "-ci{0:00000}" -f [int]$env:build_number
+}
+
+$assembly_version = "$major.$minor.0";
```

Comment:
I wonder if making the assembly version "M.0.0.0" is a better idea, otherwise it becomes a breaking change for the .NET Framework since it is strong named. Realised this when I saw Serilog does that.